coordinator-service/src/app.ts

```diff
@@ -146,6 +176,47 @@ export function initExpress({
         },
     )
 
+    app.post(
+        '/attest',
+        authenticateRequests,
+        allowParticipants,
+        bodyParser.json(),
+        (req, res) => {
+            const participantId = req.participantId
+            logger.debug(`POST /attest ${participantId}`)
+            try {
+                const body = req.body
+                if (!isSignedData(body)) {
+                    throw new Error(
+                        `Body for attestation by participant ${participantId} should have been signed data: ${JSON.stringify(
+                            body,
+                        )}`,
+                    )
+                }
+                const { data, signature } = body
+                if (
+                    !authenticateStrategy.verifyMessage(
+                        data,
+                        signature,
+                        participantId,
+                    )
+                ) {
+                    throw new Error(
+                        `Could not verify signed data for attestation by participant ${participantId}`,
+                    )
+                }
+                coordinator.addAttestation(data as Attestation, participantId)
```

Comment:
instead of directly using `data as Attestation`, use a type guard (https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards). This will also allow us to error instead of a bad attestation. We already use it in a few places in the code, you can follow those examples.